runtime/js/10_permissions.js

```diff
@@ -122,12 +122,18 @@
   function cache(desc, state) {
     let { name: key } = desc;
     if (
-      (desc.name === "read" || desc.name === "write") &&
+      (desc.name === "read" || desc.name === "write" || desc.name === "ffi") &&
       ReflectHas(desc, "path")
     ) {
       key += `-${desc.path}`;
     } else if (desc.name === "net" && desc.host) {
       key += `-${desc.host}`;
+    } else if (desc.name === "run" && desc.command) {
+      key += `-${desc.command}`;
+    } else if (desc.name === "env" && desc.variable) {
+      key += `-${desc.variable}`;
+    } else {
+      key += "$";
```

Comment:
Is this necessary?

Reply:
This is so cache keys can't conflict e.g. for descriptors `{ name: "read-foo" }` and `{ name: "read", path: "foo" }`. Permission name validation already prevents that first descriptor but this code should be more robust

Reply:
Ok. Makes sense